aptos-move/framework/aptos-framework/sources/resource_account.move

```diff
@@ -181,11 +223,23 @@ module aptos_framework::resource_account {
 
         let resource = account::create_signer_with_capability(&resource_signer_cap);
         account::rotate_authentication_key_internal(&resource, ZERO_AUTH_KEY);
-        resource_signer_cap
+
+        let data = if (exists<BlobStore>(source_addr)) {
+            let blob_store = borrow_global_mut<BlobStore>(source_addr);
+            if (table::contains(&blob_store.store, resource_addr)) {
+                table::remove(&mut blob_store.store, resource_addr)
```

Comment:
I still don't fully understand this feature. Why not provide access to the data in the blob store? Why would anyone want to extract the data while rotating the key to 0x0?

Reply:
This is expected to be called on `init_module` when you extract your signer cap and ... well the data for the module.

Reply:
Makes sense now.